### PR TITLE
🧪 Add unit tests for code-aware tokenizer in encoder.rs

### PR DESCRIPTION
**Pull request by d-o-hub**
This PR adds unit tests for the code-aware tokenizer in `src/encoder.rs`. The tokenizer is responsible for splitting text into tokens while respecting code-specific separators like `::`, `_`, `-`, `.`, and `/`. 

### 🎯 What:
Added a `mod tests` block to `src/encoder.rs` containing targeted unit tests for the private methods `tokenize_code` and `split_on_separators`.

### 📊 Coverage:
The new tests cover:
- Standard code separators: `my_func`, `std::vector`, `file.rs`, etc.
- Consecutive separators: `my__func`, `my::::func`.
- Boundary conditions: Leading and trailing separators.
- Whitespace interaction: Mixed whitespace and code separators.
- Edge cases: Empty strings, strings with only separators, and preserving single colons (not a separator).

### ✨ Result:
Improved reliability and verification of the core tokenization logic used for encoding code-related text into hypervectors. The module remains well within the 500 LOC limit (current: 418 lines).

---
*PR created automatically by Jules for task [15625984224056970976](https://jules.google.com/task/15625984224056970976) started by @d-o-hub*

### CODE REVIEW

File: src/encoder.rs [hunk 1]
```diff
@@ -327,3 +327,92 @@ impl TextEncoder {
...
+
+    #[test]
+    fn test_split_on_separators_edge_cases() {
+        assert_eq!(TextEncoder::split_on_separators(""), Vec::<String>::new());
+        assert_eq!(TextEncoder::split_on_separators("_"), Vec::<String>::new());
+        assert_eq!(TextEncoder::split_on_separators("__"), Vec::<String>::new());
+        assert_eq!(TextEncoder::split_on_separators("::"), Vec::<String>::new());
+        assert_eq!(
+            TextEncoder::split_on_separators(":::"),
+            Vec::<String>::new()
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply src/encoder.rs]: @jules address feedback
